internal/pkg/term/progress/cloudformation.go

```diff
@@ -149,22 +149,24 @@ type ecsServiceResourceComponent struct {
 // Listen creates deploymentRenderers if the service is being created, or updated.
 // It closes the Done channel if the CFN resource is Done and the deploymentRenderers are also Done.
 func (c *ecsServiceResourceComponent) Listen() {
-	var prevStatus string
 	renderers := []DynamicRenderer{c.resourceRenderer}
 	for ev := range c.cfnStream {
 		if c.logicalID != ev.LogicalResourceID {
 			continue
 		}
-		curStatus := ev.ResourceStatus
-		if curStatus != prevStatus && cloudformation.StackStatus(curStatus).UpsertInProgress() {
+		if cloudformation.StackStatus(ev.ResourceStatus).UpsertInProgress() {
+			if ev.PhysicalResourceID == "" {
+				// New service creates receive two "CREATE_IN_PROGRESS" events.
+				// The first event doesn't have a service name yet, the second one has.
```

Comment:
Hmm is this behavior documented anywhere?

Reply:
I couldn't find anything in official docs 🤷  but from observation, it seems we first go into `CREATE_IN_PROGRESS` when there is no service yet, then a second event occurs which actually calls [`CreateService`](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_CreateService.html) and returns to us the service ARN:
![Screen Shot 2021-01-27 at 5 20 38 PM](https://user-images.githubusercontent.com/879348/106076224-ffb6ee80-60c3-11eb-98ea-c3aa444764c2.png)

Reply:
yeah seems like it applies to the other resources creation as well/